File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-0.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @param {number} k
     * @return {number}
     */
    kthSmallest(root, k) {
        const arr = [];
        const postOrderTraversal = function(node, arr){
            if(!node){
                return;
            }
            postOrderTraversal(node.left, arr);
            arr.push(node.val);
            postOrderTraversal(node.right, arr);
        }
      
        postOrderTraversal(root, arr);
        return arr[k - 1];
    }
}

File: Data Structures & Algorithms/lowest-common-ancestor-in-binary-search-tree/submission-0.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

public class Solution {

    public TreeNode LowestCommonAncestor(TreeNode root, TreeNode p, TreeNode q) {
        TreeNode ans = null;
        if((root.val > p.val && root.val < q.val) || (root.val < p.val && root.val > q.val)){
            ans = root;
        }
        else if(q.val > root.val && p.val > root.val){
            ans = LowestCommonAncestor(root.right, q, p);
        }
        else if(q.val < root.val && p.val < root.val){
            ans = LowestCommonAncestor(root.left, q, p);
        }
        else if(root.val == q.val){
            ans = q;
        }
        else if(root.val == p.val){
            ans = p;
        }
        return ans;
    }  
  
}
